monitoring/nethealth.go

```diff
@@ -396,6 +409,30 @@ func getPeerName(labels []*dto.LabelPair) (peer string, err error) {
 	return "", trace.NotFound("unable to find %s label", peerLabel)
 }
 
+// filterNetData filters the networkData. Nethealth may retain metrics for nodes
+// that have previously left the cluster. This data is now irrelevant and should
+// not be further processed.
+func (c *nethealthChecker) filterNetData(netData map[string]networkData) (filtered map[string]networkData, err error) {
+	members, err := c.SerfClient.Members()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return filterBySerf(netData, members)
+}
+
+// filterBySerf removes netData for nodes that are not part of the serf cluster.
+func filterBySerf(netData map[string]networkData, members []serf.Member) (filtered map[string]networkData, err error) {
+	filtered = make(map[string]networkData)
+	for _, member := range members {
+		if data, exists := netData[member.Addr.String()]; exists {
```

Comment:
We should be filtering out not just those that don't exist anymore. But those that are in left/failed state as well (or just "!= alive" for simplicity). If you look at the bug report, you will see that the member is actually a part of "serf members" but in a "left" state.

Reply:
Ah, you're right. Added check for serf member status. 0d27f7c